src/main/java/com/icthh/xm/uaa/service/UserService.java

```diff
@@ -154,14 +154,14 @@ public User createUser(UserDTO user) {
     public Optional<UserDTO> updateUser(UserDTO updatedUser) {
         return userRepository.findById(updatedUser.getId())
             .map(user -> {
+                //use blockUserAccount/activateUserAccount
+                user.setActivated(user.isActivated());
+                //use changeUserRole
+                user.setRoleKey(user.getRoleKey());
                 user.setFirstName(updatedUser.getFirstName());
                 user.setLastName(updatedUser.getLastName());
                 user.setLangKey(updatedUser.getLangKey());
                 user.setImageUrl(updatedUser.getImageUrl());
-                user.setActivated(updatedUser.isActivated());
-                if (StringUtils.isNoneBlank(updatedUser.getRoleKey())) {
```

Comment:
it can broke current tenants
like update all fields but not send rolle

Reply:
yes, it may) but this is actually main purpose of this PR.
**userState** and **userRole** life cycle is slightly different from Lang or ImageUrl
and in case of general data update, user role could be modified by hacker

KafkaEvent Constants.UPDATE_ACCOUNT_EVENT_TYPE is using userService.saveUser(user) to update user data, this method is not changed and will update all data (also is bad practice)

i definitely need help to identify other places and provide best approach to this CR

thnx!

Reply:
we using this method on many tenants
maybe 3 or 4 tenant update role and activated using this method

Reply:
Restored old behaviour with following rules
1) fore new tenants by default new behaviour will be applied 
default-uaa.yml
strictUserManagement: true
2) for existing old behaviour with log.warn is used